command/agent.go

```diff
@@ -805,25 +775,47 @@ func (c *AgentCommand) Run(args []string) int {
 		close(c.startedCh)
 	}
 
-	// Listen for signals
-	// TODO: implement support for SIGHUP reloading of configuration
-	// signal.Notify(c.signalCh)
-
 	var g run.Group
 
+	g.Add(func() error {
+		for {
+			select {
+			case <-c.SighupCh:
+				c.UI.Output("==> Vault Agent config reload triggered")
+				err := c.reloadConfig(c.flagConfigs)
+				if err != nil {
+					c.outputErrors(err)
+				}
+				// Send the 'reloaded' message on the relevant channel
+				select {
+				case c.reloadedCh <- struct{}{}:
+				default:
+				}
+			case <-ctx.Done():
+				c.notifySystemd(systemd.SdNotifyStopping)
```

Comment:
We don't need multiple notifySystemd calls all over the place.  In fact I'd remove all the existing ones, and replace them with a single one at the end of this func, after g.Run returns.

Reply:
Remove all existing: `systemd.SdNotifyStopping` ones? (just double checking, we'd leave the reloading/ready etc in situ).

Reply:
That's right - all existing ones in Agent code.